cmd/ctl/pkg/create/certificaterequest/certificaterequest.go

```diff
@@ -245,7 +245,7 @@ func (o *Options) Run(ctx context.Context, args []string) error {
 	if o.FetchCert {
 		fmt.Fprintf(o.ErrOut, "CertificateRequest %v in namespace %v has not been signed yet. Wait until it is signed...\n",
 			req.Name, req.Namespace)
-		err = wait.Poll(time.Second, o.Timeout, func() (done bool, err error) {
+		err = wait.PollUntilContextTimeout(ctx, time.Second, o.Timeout, true, func(ctx context.Context) (done bool, err error) {
```

Comment:
suggestion: From looking at https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#Poll it seems that `Poll` doesn't immediately poll. As such this change (and others like it) appears to be a change in behaviour.

> Poll always waits the interval before the run of 'condition'. 'condition' will always be invoked at least once. 

Is that intentional? If not, should this be:

```suggestion
		err = wait.PollUntilContextTimeout(ctx, time.Second, o.Timeout, false, func(ctx context.Context) (done bool, err error) {
```

Reply:
Indeed, this is a behavioral change.
However, I think that we were often incorrectly using `Poll` instead of `PollImmediate`.
I propose that we make the breaking change for all testing code (mostly just loops waiting for a condition) & I'll create a new PR to change the behavior for the non-testing code where it makes sense.